### PR TITLE
Feature/plan oic 56398 openapi path selector

### PR DESCRIPTION
**Pull request by paullanoracle**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -225,6 +233,11 @@
           "when": "resourceDirname =~ /(\\/|\\\\)api$/ && resourceExtname == .json",
           "command": "orab.add.convert",
           "group": "orab@3"
+        },
+        {
+          "when": "resourceDirname =~ /(\\/|\\\\)api$/ && resourceExtname == .json",
+          "command": "orab.add.convert.append",
+          "group": "orab@3"
```
Comment (sebastien-ma) [inline package.json:240]: Can you change the order to 4?

File: src/api.ts [hunk 2]
```diff
@@ -398,22 +403,33 @@ export namespace conversion {
...
+    }
+
+    return callAPI(() => client.post(endpoint, form) as Promise<AxiosResponse<OpenAPINS.Root>>, (err) => {
       if (err instanceof AxiosError && err.response?.status !== 404) {
-        log.infoServer(err.response?.data);
+        logInfoServer(err.response?.data);
       }
-      throw new RABError(`Failed to call 'POST ${endpoint}'`, err);
-    }
+      showErrorMessage("❌ Conversion failed");
```
Comment (sebastien-ma) [inline src/api.ts:431]: Let's try to decouple api layer from UI interaction. Throwing the error out and let the upper layer decide how to handle it.

File: src/commands/convert-openapi-document.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright © 2023, Oracle and/or its affiliates.
+ * This software is licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as vscode from 'vscode';
+
+import * as _fs from 'fs';
+
+import { bindNodeCallback, catchError, filter, firstValueFrom, from, map, switchMap, tap, throwError } from 'rxjs';
```
Comment (sebastien-ma) [inline src/commands/convert-openapi-document.ts:10]: We discussed this before. Do you plan to move away from rxjs?

File: src/extension.ts [hunk 4]
```diff
@@ -23,18 +23,16 @@ import * as insertSecurityPolicy from './commands/insert-security-policy';
...
 import * as addOutlineProvider from './providers/add-outline-provider';
 import * as postmanSelectRequrestWebview from './providers/copilot-webview';
 import * as profileListProvider from './providers/profile-list-provider';
 
-import { init as initProfileManager, get as getProfileManager } from './profile-manager-provider';
+import { get as getProfileManager, init as initProfileManager } from './profile-manager-provider';
 
 async function registerCommands(context: vscode.ExtensionContext) {
   addLocate.register(context);
-  convertADD.register(context);
```
Comment (sebastien-ma) [inline src/extension.ts:37]: why remove this line?

**Review by sebastien-ma — CHANGES_REQUESTED** [review]

File: src/commands/convert-openapi-document.ts (hunk 3 above)
Comment (paullanoracle) [reply src/commands/convert-openapi-document.ts:10]: This is not for API calls, but mostly UI works.

File: src/api.ts (hunk 2 above)
Comment (paullanoracle) [reply src/api.ts:431]: The error is shown in another place. We didn't hide it.

File: src/extension.ts (hunk 4 above)
Comment (sebastien-ma) [reply src/extension.ts:37]: Can you please follow the same convention?

**Review by sebastien-ma — APPROVED** [review]
Approved. But let's not introduce any new Rxjs constructs. It should be phased out in future.

**Review by tacobacca — APPROVED** [review]

**Comment by paullanoracle** [comment]
Hold on merging due to backend is not ready.

**Comment by paullanoracle** [comment]
Backend is ready in dev pipeline (promoted).

**Pull request merged**